### PR TITLE
chore(docs): What is CDKTF and Interoperability

### PR DESCRIPTION
**Pull request by laurapacilio**
In this PR I:
- Copyedited the What is CDKTF and HCL Interoperability pages
- Added more context at the top of the HCL interoperability page for new users
- Shortened the title of the HCL Interoperability page and also the headings. The goal is to make them easier to read and more concise, but lmk if you disagree!

Thank you :-)

### CODE REVIEW

File: website/docs/cdktf/index.html.md
Comment (danieldreier) [inline website/docs/cdktf/index.html.md:36]: This feels _slightly_ too strong to me - can I negotiate this down to simply "You are comfortable doing your own troubleshooting and do not require commercial support"? The "be careful" type warnings really scare people off.

File: website/docs/cdktf/index.html.md
Comment (danieldreier) [inline website/docs/cdktf/index.html.md:40]: Is this section supposed to have the copy from [the readme](https://github.com/hashicorp/terraform-cdk#project-maturity-and-production-readiness)?

File: website/docs/cdktf/index.html.md
Comment (danieldreier) [inline website/docs/cdktf/index.html.md:35]: maybe link this to the constructs documentation, and then make sure we cover "why" as well as "how" in there?

**Review by danieldreier — CHANGES_REQUESTED** [review]
This looks great - these improvements are great. 👍 pending a few small suggestions

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md:36]: yes! I'll push up that change

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md:35]: yes! I'll add a link but..can you please just finish this sentence: "For example, you want to create constructs (link) to ......" I just think this is the time to emphasize the power of CDKTF to get people interested and show how it can solve problems. And so what's something someone might really be like OH WOW I want to make a construct for that!

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md:40]: Um yes - I think it's appropriate to say here that this under active development and so I put this here. We do something similar in the plugin provider framework docs: https://www.terraform.io/docs/plugin/framework/index.html. If you disagree with having this here, let me know! Perhaps @schersh can also weigh in on this!

**Comment by laurapacilio** [comment]
@danieldreier I didn't mean to push the commands page to this branch but alas, I have. So....you can review that too. I'm sorry about that, but then I guess we can check it off our list!

File: website/docs/cdktf/index.html.md
Comment (schersh) [reply website/docs/cdktf/index.html.md:40]: @laurapacilio I think it would be best if the #Product Maturity section in docs was the same as the content we have in the README (https://github.com/hashicorp/terraform-cdk#project-maturity-and-production-readiness) the README copy has already been vetted internally, and we should be consistent with the wording in both places. Does that sound okay to you?

Comment (schersh) [reply website/docs/cdktf/index.html.md:40]: My only other question is this section, is if it would be worthwhile linking to our forum & issues since we're asking for feedback here? 

Something like: 
You can drop notes in our [discussion forum](https://discuss.hashicorp.com/c/terraform-core/cdk-for-terraform/47), and report any issues you encounter in our [issue tracker](https://github.com/hashicorp/terraform-cdk/issues/new/choose).

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (schersh) [inline website/docs/cdktf/cli-reference/commands.html.md:416]:
```suggestion
An example use case for watch could be:
```
I think we could simplify the language here

File: website/docs/cdktf/concepts/interoperability-workflows.html.md
Comment (schersh) [inline website/docs/cdktf/concepts/interoperability-workflows.html.md:5]:
```suggestion
description: "Use configurations written in HCL and configurations written in CDK for Terraform together to define and provision infrastructure."
```

slightly stronger for SEO to lead with the verb and avoid "learn to, etc."

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (schersh) [inline website/docs/cdktf/cli-reference/commands.html.md:317]: I think we need some clarification here that it's the `--from-terraform-project` flag that is only supported for Typescript (I had to read this a few times to understand this statement wasn't referring to the `init` command

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (schersh) [inline website/docs/cdktf/cli-reference/commands.html.md:401]: just curious @danieldreier but is there any flag for using `watch` that will prevent it from deploying changes without asking for permission? That seems like something maybe worth calling out with a *warning*?

File: website/docs/cdktf/index.html.md
Comment (schersh) [inline website/docs/cdktf/index.html.md:34]:
```suggestion
- You have a strong preference or need to use a procedural language to define infrastructure.
```

**Review by schersh — COMMENTED** [review]
Thank you @laurapacilio! I added a few comments throughout, but I think these pages are looking great overall.

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (laurapacilio) [reply website/docs/cdktf/cli-reference/commands.html.md:401]: yeah you know what this is on me - I combined this in with the text and now users may miss it.... I'm going to try combining it with the warning we already have above since we probably want to avoid a bunch of warning blocks.  But @danieldreier let us know about the flag as well - if there is one, I can add that too!

File: website/docs/cdktf/concepts/interoperability-workflows.html.md
Comment (laurapacilio) [reply website/docs/cdktf/concepts/interoperability-workflows.html.md:5]: oh I didn't know that! I'll keep that in mind for the future - thank you!

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md:40]: oh wow I didn't realize the README was official copy. I'll change it to be exactly that. And I can add a callout to the forum and issue tracker - great idea!

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (laurapacilio) [reply website/docs/cdktf/cli-reference/commands.html.md:317]: Thank you! I will push up a fix for this

File: website/docs/cdktf/index.html.md
Comment (danieldreier) [reply website/docs/cdktf/index.html.md]: `for example, you want to create constructs to model a reusable infrastructure pattern composed of multiple resources and convenience methods.`

**Comment by laurapacilio** [comment]
Okay @schersh and @danieldreier I pushed up changes that I think address all of Sarah's comments (thank you again!). Let me know if you see anything else.

File: website/docs/cdktf/index.html.md
Comment (danieldreier) [reply website/docs/cdktf/index.html.md]: @schersh FWIW I added some very similar language to the community doc https://github.com/hashicorp/terraform-cdk/pull/1066

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md]: perfect! I pushed this up :-)

File: website/docs/cdktf/index.html.md
Comment (schersh) [reply website/docs/cdktf/index.html.md]: do you think it's overkill to have it in both places? I think it's nice to add the destination wherever we say we want feedback

Comment (laurapacilio) [reply website/docs/cdktf/index.html.md]: @schersh In a recent commit, I added a link to the community page Daniel made! I agree that we should provide some type of link to guide folks - and I was thinking this could be a good compromise between not having anything and repeating content that's on the community page. Check it out in the most recent draft and let me know your thoughts on this approach!

Comment (schersh) [reply website/docs/cdktf/index.html.md]: Perfect, thanks @laurapacilio  - that sounds good to me!

File: website/docs/cdktf/index.html.md
Comment (schersh) [inline website/docs/cdktf/index.html.md:14]: I get a 404 when I try to access this terraform platform image https://github.com/hashicorp/terraform-cdk/blob/main/website/docs/cdktf/terraform-platform.png
maybe it hasn't been moved from it's prior location yet? 
@laurapacilio

**Review by schersh — APPROVED** [review]
I'm approving with two comments/caveats: 

- Need to verify the file location of the terraformplatform.png
- Still a "TODO" item listed for the `watch` command. 

Thanks!

File: website/docs/cdktf/index.html.md
Comment (laurapacilio) [reply website/docs/cdktf/index.html.md:14]: oh! For a second I was stumped, but you know what it is? The path you sent is on main! and this image is definitely not in that location on main. Try this: https://github.com/hashicorp/terraform-cdk/blob/terraform-io-docs/website/docs/cdktf/terraform-platform.png

**Comment by laurapacilio** [comment]
@danieldreier Let us know what you think about the `watch` command TODO (above) and if you feel comfortable approving. Once that's taken care of, I'll merge it in! :-)

**Review by danieldreier — APPROVED** [review]

File: website/docs/cdktf/cli-reference/commands.html.md
Comment (skorfmann) [inline website/docs/cdktf/cli-reference/commands.html.md:422]: I think `--stack` is just a positional argument according to the help output. `auto-approve` behaves similar to TF Core and will apply changes without additional confirmation. `auto-approve` is mandatory right now.

```
cdktf watch [stack] [OPTIONS]

[experimental] Watch for file changes and automatically trigger a deploy

Positionals:
  stack  Deploy stack which matches the given id only. Required when more than one stack is present in the app                                                                                                                                                                                                                                                      [string]

Options:
      --version                   Show version number                                                                                                                                                                                                                                                                                                              [boolean]
      --disable-logging           Dont write log files. Supported using the env CDKTF_DISABLE_LOGGING.                                                                                                                                                                                                                                             [boolean] [default: true]
      --disable-plugin-cache-env  Dont set TF_PLUGIN_CACHE_DIR automatically. This is useful when the plugin cache is configured differently. Supported using the env CDKTF_DISABLE_PLUGIN_CACHE_ENV.                                                                                                                                             [boolean] [default: false]
      --log-level                 Which log level should be written. Only supported via setting the env CDKTF_LOG_LEVEL                                                                                                                                                                                                                                             [string]
  -a, --app                       Command to use in order to execute cdktf app                                                                                                                                                                                                                                                                                    [required]
  -o, --output                    Output directory                                                                                                                                                                                                                                                                                         [required] [default: "cdktf.out"]
      --auto-approve              Auto approve                                                                                                                                                                                                                                                                                                    [boolean] [default: false]
  -h, --help                      Show help
```

Comment (laurapacilio) [reply website/docs/cdktf/cli-reference/commands.html.md:446]: okay! I added in this help text, added in an explanation for the example, and fussed with the formatting a bit to make it match the other sections on the page.  I'm gonna push and we can make any final tweaks in the pass I want to make before Hashiconf :-)

**Pull request merged**
